http/okhttp/src/main/java/com/okta/commons/http/okhttp/OkHttpRequestExecutorFactory.java

```diff
@@ -20,15 +20,32 @@
 import com.okta.commons.http.RequestExecutorFactory;
 import com.okta.commons.http.RetryRequestExecutor;
 import com.okta.commons.http.config.HttpClientConfiguration;
+import okhttp3.OkHttpClient;
 
 /**
  * @since 1.2.0
  */
 @AutoService(RequestExecutorFactory.class)
 public class OkHttpRequestExecutorFactory implements RequestExecutorFactory {
 
+    private final OkHttpClient client;
+
+    public OkHttpRequestExecutorFactory() {
+        this(null);
+    }
+
+    public OkHttpRequestExecutorFactory(OkHttpClient client) {
```

Comment:
```suggestion
    /**
     * Creates a `OkHttpRequestExecutorFactory` that create `OkHttpRequestExecutor` instances that 
     * use a shared `OkHttpClient`.
     *
     * @param client a custom configured `OkHttpClient`.
     * @since 1.3.0
     */
    public OkHttpRequestExecutorFactory(OkHttpClient client) {
```

A quick pass at adding Javadoc, made me think about a few downsides here, that _might_ not be worth it (or may require a more robust solution).

Retries are handled with custom logic, time-out configuration would be skipped, the proxy config would be ignored, I don't remember if any cookies are actually set by Okta, but they were disabled as they were not used by the client.

https://github.com/okta/okta-commons-java/blob/9631609446fd4b4dcc9cee5f0fd0c4dea289eba4/http/okhttp/src/main/java/com/okta/commons/http/okhttp/OkHttpRequestExecutor.java#L70-L93

IIRC, in the case of Android, the proxy configuration _should_ get picked up automatically.  Time-out's _could_ be a note in the Javadoc (though that might not be obvious).  The retry issue though may be the cause of some unexpected behavior.  `OkHttpClient` will retry requests (IIRC sometimes in parallel), then the SDK will attempt to retry any failed requests, which in turn would cause the `OkHttpClient` to retry requests again.  This could result in slow requests and increased rate limit usage.